sechub-adapter/src/main/java/com/daimler/sechub/adapter/AbstractWebScanAdapterConfigBuilder.java

```diff
@@ -3,160 +3,191 @@
+
+                    public FormScriptLoginActionBuilder description(String description) {
+                        action.description = description;
+                        return this;
+                    }
+
+                    public FormScriptLoginActionBuilder unit(SecHubTimeUnit unit) {
+                        action.unit = unit;
+                        return this;
+                    }
+
+                    public FormScriptLoginPageBuilder endStep() {
+                        page.getActions().add(action);
+                        return FormScriptLoginPageBuilder.this;
+                    }
+
+                }
+            }
+        }
+
+        private void doEndLogin(AbstractLoginConfig loginConfig) {
+            createdLoginConfig = loginConfig;
+        }
+
+        public class FormAutoDetectLoginBuilder {
+
+            private FormAutoDetectLoginConfig formAutomatedLoginConfig = new FormAutoDetectLoginConfig();
+
+            public FormAutoDetectLoginBuilder username(String user) {
+                formAutomatedLoginConfig.user = encrypt(user);
+                return this;
+            }
+
+            public FormAutoDetectLoginBuilder password(String password) {
+                formAutomatedLoginConfig.password = encrypt(password);
+                return this;
+            }
+
+            @SuppressWarnings("unchecked")
+            public final B endLogin() {
+                doEndLogin(formAutomatedLoginConfig);
+                return (B) AbstractWebScanAdapterConfigBuilder.this;
+            }
+
+        }
+
+        public class BasicLoginBuilder {
+
+            private BasicLoginConfig basicLoginConfig = new BasicLoginConfig();
+
+            public BasicLoginBuilder username(String user) {
+                basicLoginConfig.user = encrypt(user);
+                return this;
+            }
+
+            public BasicLoginBuilder password(String password) {
+                basicLoginConfig.password = encrypt(password);
+                return this;
+            }
+
+            public BasicLoginBuilder realm(String realm) {
+                basicLoginConfig.realm = encrypt(realm);
+                return this;
+            }
+
+            @SuppressWarnings("unchecked")
+            public final B endLogin() {
+                doEndLogin(basicLoginConfig);
+                return (B) AbstractWebScanAdapterConfigBuilder.this;
+            }
+        }
+
+    }
+
+    /**
+     * @return a new login builder when not already started or former login was
+     *         ended. Otherwise current login builder is returned
+     */
+    public LoginBuilder login() {
+        currentLoginBuilder = new LoginBuilder();
+        return currentLoginBuilder;
+    }
+
+    @Override
+    void packageInternalCustomBuild(C config) {
+        config.maxScanDuration = maxScanDuration;
+
+        if (currentLoginBuilder == null) {
+            return;
+        }
+        if (currentLoginBuilder.createdLoginConfig == null) {
```

Comment:
An `OR` does not work in this case. I can only use an `AND`.

Reply:
I think you missed this in your changes.

Reply:
I think you missed this in your changes.

Reply:
No, I changed it to `AND` and it is not working. In addition, it does not provide any benefit. It is easier to read, having both statements separated.

Reply:
Ok, that's strange. That's why I thought about `OR`. With an OR it would exit the validation early if the first condition is true without interpreting the second one. But if the first one is false and the second one true it would still exit, without the risk of accessing a null-pointer at any time.

```
if (currentLoginBuilder == null || currentLoginBuilder.createdLoginConfig == null) {
            return;
}
```

Reply:
You are correct about the `OR`. Regardless, in my opinion using an `OR` will not improve the readability. Therefore, I will not change it.